---
int64 bits_x
int64 bits_y 
int64 bits_z